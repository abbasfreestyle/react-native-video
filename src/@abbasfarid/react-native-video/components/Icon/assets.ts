import FastForward from '../../assets/fast-forward.svg';
import Mute from '../../assets/mute.svg';
import Pause from '../../assets/pause.svg';
import Play from '../../assets/play.svg';
import Rewind from '../../assets/rewind.svg';
import Sound from '../../assets/sound.svg';

export const svgIcons = {
  FastForward,
  Mute,
  Pause,
  Play,
  Rewind,
  Sound,
};

export type SvgAssets = keyof typeof svgIcons;
