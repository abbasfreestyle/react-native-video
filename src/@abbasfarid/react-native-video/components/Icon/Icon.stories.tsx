import React from 'react';
import { ScrollView, StyleSheet, Text, View } from 'react-native';

import { storiesOf } from '@storybook/react-native';

import { SvgAssets, svgIcons } from './assets';
import { Icon } from './Icon';

const styles = StyleSheet.create({
  wrapper: {
    alignItems: 'center',
    flexDirection: 'row',
    height: 50,
    justifyContent: 'space-between',
    paddingHorizontal: 50,
  },
});

const stories = storiesOf('Components', module);

stories.add(`SVG Icons`, () => (
  <ScrollView>
    {Object.keys(svgIcons).map((type) => (
      <View key={type} style={styles.wrapper}>
        <Text>{type}</Text>
        <Icon size={25} type={type as SvgAssets} />
      </View>
    ))}
  </ScrollView>
));
