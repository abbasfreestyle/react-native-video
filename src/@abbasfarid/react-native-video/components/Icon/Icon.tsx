import React from 'react';

import { SvgProps } from 'react-native-svg';

import { SvgAssets, svgIcons } from './assets';

interface Props extends Omit<SvgProps, 'height' | 'width'> {
  size?: number;
  type: SvgAssets;
}

export const Icon = ({ size = 20, type, ...props }: Props) => {
  const SvgIcon = svgIcons[type];

  return <SvgIcon height={size} width={size} {...props} />;
};
