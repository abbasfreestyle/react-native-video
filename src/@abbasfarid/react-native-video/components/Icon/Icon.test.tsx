import React from 'react';

import { render } from '@testing-library/react-native';

import { SvgAssets, svgIcons } from './assets';
import { Icon } from './Icon';

// Automatic iteration over all Icons and taking snapshots
describe('Icon', () => {
  Object.keys(svgIcons).map((type) => {
    test(`matches the snapshot ${type}`, () => {
      const { toJSON } = render(<Icon type={type as SvgAssets} />);
      expect(toJSON()).toMatchSnapshot();
    });
    return null;
  });
});
