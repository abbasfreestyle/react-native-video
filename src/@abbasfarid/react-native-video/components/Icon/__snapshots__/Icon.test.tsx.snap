// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon matches the snapshot 1`] = `
<SvgMock
  height={20}
  width={20}
/>
`;

exports[`Icon matches the snapshot 2`] = `
<SvgMock
  height={20}
  width={20}
/>
`;

exports[`Icon matches the snapshot 3`] = `
<SvgMock
  height={20}
  width={20}
/>
`;

exports[`Icon matches the snapshot 4`] = `
<SvgMock
  height={20}
  width={20}
/>
`;

exports[`Icon matches the snapshot 5`] = `
<SvgMock
  height={20}
  width={20}
/>
`;

exports[`Icon matches the snapshot 6`] = `
<SvgMock
  height={20}
  width={20}
/>
`;
